main/pipeline/pipeline.h

```diff
@@ -48,6 +48,21 @@ size_t partitionPackageFiles(const core::GlobalState &gs, absl::Span<core::FileR
 void unpartitionPackageFiles(std::vector<ast::ParsedFile> &packageFiles,
                              std::vector<ast::ParsedFile> &&nonPackageFiles);
 
+struct CondensationLayerInfo {
+    // The `__package.rb` sources in this layer of the condensation graph.
+    absl::Span<ast::ParsedFile> packageFiles;
+
+    // The ruby sources in this layer of the condensation graph.
+    absl::Span<core::FileRef> sourceFiles;
+};
+
+// Using the condensation graph, sort the package and source files according to the layer they would show up in the
+// parallel traversal of the condensation graph. The `packageFiles` vector will be mutated to include non-test versions
+// of the package files included originally.
+std::vector<CondensationLayerInfo> condensationLayers(const core::GlobalState &gs,
```

Comment:
We're saying that a "layer" is "all SCCs at a given depth a traversal of the condensation graph"? or is a layer an SCC?